spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+          .build();
+
+  public static final ConfigOption<Boolean> LEADER_ELECTION_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.leaderElection.enabled")
+          .defaultValue(false)
+          .description(
+              "Enable leader election for the operator to allow running standby instances.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Boolean.class)
+          .build();
+
+  public static final ConfigOption<String> LEADER_ELECTION_LEASE_NAME =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.leaderElection.leaseName")
+          .defaultValue("spark-operator-lease")
+          .description(
+              "Leader election lease name, must be unique for leases in the same namespace.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(String.class)
+          .build();
+
+  public static final ConfigOption<Long> LEADER_ELECTION_LEASE_DURATION_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.leaderElection.leaseDurationSeconds")
+          .defaultValue(1200L)
+          .description("Leader election lease duration.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> LEADER_ELECTION_RENEW_DEADLINE_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.leaderElection.renewDeadlineSeconds")
+          .defaultValue(600L)
+          .description("Leader election renew deadline.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> LEADER_ELECTION_RETRY_PERIOD_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.leaderElection.retryPeriodSeconds")
+          .defaultValue(180L)
+          .description("Leader election retry period.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static LeaderElectionConfiguration getLeaderElectionConfig() {
+    return new LeaderElectionConfiguration(
+        LEADER_ELECTION_LEASE_NAME.getValue(),
+        OPERATOR_NAMESPACE.getValue(),
+        Duration.ofSeconds(LEADER_ELECTION_LEASE_DURATION_SECONDS.getValue()),
+        Duration.ofSeconds(LEADER_ELECTION_RENEW_DEADLINE_SECONDS.getValue()),
+        Duration.ofSeconds(LEADER_ELECTION_RETRY_PERIOD_SECONDS.getValue()));
+  }
+
+  public static GenericRetry getOperatorRetry() {
+    GenericRetry genericRetry =
+        new GenericRetry()
+            .setMaxAttempts(RETRY_MAX_ATTEMPTS.getValue())
+            .setInitialInterval(
+                Duration.ofSeconds(RETRY_INITIAL_INTERVAL_SECONDS.getValue()).toMillis())
+            .setIntervalMultiplier(RETRY_INTERVAL_MULTIPLIER.getValue());
+    if (RETRY_MAX_INTERVAL_SECONDS.getValue() > 0) {
```

Comment:
Could you add a comment about `else` case? I'm wondering what is the default value when a user gives `0` or negative.